eureka2-core/src/main/java/com/netflix/eureka2/interests/ChangeNotifications.java

```diff
@@ -158,4 +196,157 @@ private ChangeNotifications() {
+     *     <li>{ Modify, Delete } = { Delete }</li>
+     *     <li>{ Delete, Add } = { Add }</li>
+     *     <li>{ Delete, Modify } = { Modify }</li>
+     * </ul>
+     */
+    public static <T> Transformer<List<List<ChangeNotification<T>>>, List<ChangeNotification<T>>> collapseLists(final Comparator<T> identityComparator) {
+        return new Transformer<List<List<ChangeNotification<T>>>, List<ChangeNotification<T>>>() {
+            @Override
+            public Observable<List<ChangeNotification<T>>> call(Observable<List<List<ChangeNotification<T>>>> listOfListObservable) {
+                return listOfListObservable.map(new Func1<List<List<ChangeNotification<T>>>, List<ChangeNotification<T>>>() {
+                    @Override
+                    public List<ChangeNotification<T>> call(List<List<ChangeNotification<T>>> notificationLists) {
+                        Map<T, Integer> markers = new TreeMap<>(identityComparator);
+                        List<ChangeNotification<T>> result = new ArrayList<ChangeNotification<T>>();
+                        for (int i = notificationLists.size() - 1; i >= 0; i--) {
+                            collapse(notificationLists.get(i), markers, result);
+                        }
+                        Collections.reverse(result);
+                        return result;
+                    }
+                });
+            }
+        };
+    }
+
+    /**
+     * Aggregate change notifications in the specified time intervals collapsing changes pertaining to same
+     * data objects.
+     */
+    public static <T> Transformer<List<ChangeNotification<T>>, List<ChangeNotification<T>>> aggregateChanges(
+            final Comparator<T> identityComparator, final long interval, final TimeUnit timeUnit, final Scheduler scheduler) {
+        return new Transformer<List<ChangeNotification<T>>, List<ChangeNotification<T>>>() {
+            @Override
+            public Observable<List<ChangeNotification<T>>> call(Observable<List<ChangeNotification<T>>> batchUpdates) {
+                return batchUpdates.buffer(interval, timeUnit, scheduler).compose(collapseLists(identityComparator));
+            }
+        };
+    }
+
+    public static <T> Transformer<List<ChangeNotification<T>>, List<ChangeNotification<T>>> aggregateChanges(
+            final Comparator<T> identityComparator, final long interval, final TimeUnit timeUnit) {
+        return aggregateChanges(identityComparator, interval, timeUnit, Schedulers.computation());
+    }
+
+    /**
+     * It is a version of {@link #aggregateChanges} method, where the first item is emitted eagerly, and aggregation
+     * is applied only afterwards. This fits the Eureka registration model, where first batch will contain the full
+     * current registry content, which is followed by live updates.
+     */
+    public static <T> Transformer<List<ChangeNotification<T>>, List<ChangeNotification<T>>> emitAndAggregateChanges(
```

Comment:
Are all the layers of aggregateChanges and collapse necessary? It seems emitAndAggregateChanges() is the only function actually used. externally.